src/actions/build/mod.rs

```diff
@@ -137,4 +138,8 @@ impl Build {
             },
         }
     }
-}
\ No newline at end of file
+
+    fn lock_previous_results(&self) -> MutexGuard<Results> {
+        self.previous_results.lock().expect("previous results lock failed")

```

Comment:
I would inline this function - I think repeating the message string is OK and makes it a bit clearer what is going on